### ISSUE TITLE
Ongoing work to extend subsonic API

### ISSUE THREAD

**Issue by Tolriq** [spl0k/supysonic#242]
Hi,

There's currently an ongoing work to try to expand the Subsonic API and improve the global ecosystem around it.
Since you have been mentioned it would be nice if you were interested in participating.

See https://support.symfonium.app/t/expanding-subsonic-api/1115 and https://support.symfonium.app/c/subsonic-api-extensions/11

**Comment by spl0k** [comment]
Interesting, I'll chime in.

**Comment by ivan-avalos** [comment]
I wouldn't trust an API initiative from a proprietary app.

**Comment by Tolriq** [comment]
What a strange point of view care to elaborate?

If you had read just a little you'd see how it's proposed and even the proposed public org own by the servers ....

**Comment by ivan-avalos** [comment]
Yes, you're right, I should have read it a bit. Although I'd still be wary. For example, the server identification proposal seems

**Comment by ivan-avalos** [comment]
Yes, you're right. I should have read it. However, I'm still wary of this. For example, the server identification proposal might encourage server developers to introduce non-standard features that clients would need to handle especially, introducing a lot of fragmentation in the ecosystem, giving dominance to the most popular implementations. I would really only stick to semver.

**Comment by Tolriq** [comment]
The thing is that the goal of this is specially to fix the current status of Subsonic. It's gone closed source and the definition is not clear.
Every server currently implementing the API have subtle differences in the handling making the life of all clients difficult.
The server identification part, is just here to help clients to deal with the current differences between servers. 

The rest and future is proposed via the extensions, allowing servers to follow the API at their pace, without having to deal with tons of changes, and not being able to add a feature until it had support for the 12 or 20 others from that new version.

Subsonic API lacks quite a few things to rivals with the other media centers now, the solution is either to bring something easy for the servers to maintain or stop using Subsonic, and use each server specific API.
Most servers already have their internal API to do things that Subsonic does not support, but that would mean, less clients, and would do exactly what you fear, generate monopoly, if only one server publish an API and have the best clients, many users will move.

**Comment by baldurmen** [comment]
There's also the OpenSubsonic effort:

https://opensubsonic.netlify.app/
https://github.com/opensubsonic/open-subsonic-api

**Comment by Tolriq** [comment]
This is actually the result of all that work ;) Not something else.
